### PR TITLE
add log.level and log.encoding to all components

### LINKED ISSUE #3440: Expose logging options like human readable verbose and json format in values.yaml
**Is your feature request related to a problem? Please describe.**
It would be very helpful to make the logs more readable and operator friendly to expose simple yaml options in the helm chart. Cert-manager does this like:
```
## -- Map of extra arguments to pass to container.
extraArgs:
  zap-time-encoding: iso8601
  loglevel: debug
```
and it's documented on their site

**Describe the solution you'd like**
`helm show values external-secrets/external-secrets` should show a default values.yaml file which shows the currently logging settings and commented out reference docs on how to enable json, human readable, and verbosity level settings

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by KyriosGN0**
## Problem Statement
Add log.level and log.encoding to all components, default is the same as documented by cli flags 

## Related Issue

Fixes #3440 

## Proposed Changes

i added a log.level and log.encoding values to the chart and set the default to 

## Checklist

- [X] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [X] All commits are signed with `git commit --signoff`
- [] My changes have reasonable test coverage
- [X] All tests pass with `make test`
- [X] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Review by moolen — COMMENTED** [review]
Great stuff, thanks for taking care of that. Could you please also update the docs here: https://external-secrets.io/latest/api/controller-options/

It would be good to add the available time encoding options as well.
Would it be worth to rename it to time-encoding? Encoding could mean json/human-readable format of the log messages

**Comment by KyriosGN0** [comment]
> Great stuff, thanks for taking care of that. Could you please also update the docs here: https://external-secrets.io/latest/api/controller-options/
> 
> It would be good to add the available time encoding options as well. Would it be worth to rename it to time-encoding? Encoding could mean json/human-readable format of the log messages

@moolen how do i update the website? i have site folder in my repo but its ignored by .gitignore

**Comment by moolen** [comment]
There is the docs dir which contains a bunch of markdown files: https://github.com/external-secrets/external-secrets/blob/main/docs/api/controller-options.md

File: docs/api/controller-options.md [hunk 1]
```diff
@@ -41,6 +42,7 @@ The core controller is invoked without a subcommand and can be configured with t
 | `--healthz-addr`           | string   | :8081                    | The address the health endpoint binds to.                                                                             |
 | `--help`                   |          |                          | help for certcontroller                                                                                               |
 | `--loglevel`               | string   | info                     | loglevel to use, one of: debug, info, warn, error, dpanic, panic, fatal                                               |
+| `--zap-time-encoding`                                  | string   | epoch                          | loglevel to use, one of: epoch, millis, nano, iso8601, rfc3339, rfc3339nano                                                                                            |
```
Comment (moolen) [inline docs/api/controller-options.md:45]:
```suggestion
| `--zap-time-encoding`                                  | string   | epoch                          | time encoding to use, one of: epoch, millis, nano, iso8601, rfc3339, rfc3339nano                                                                                            |
```

File: docs/api/controller-options.md [hunk 2]
```diff
@@ -57,6 +59,7 @@ The core controller is invoked without a subcommand and can be configured with t
...
 | `--loglevel`           | string   | info                                  | loglevel to use, one of: debug, info, warn, error, dpanic, panic, fatal                                                                                                                                                                                                                                                                                                                                                  |
+| `--zap-time-encoding`                                  | string   | epoch                          | loglevel to use, one of: epoch, millis, nano, iso8601, rfc3339, rfc3339nano                                                                                            |
```
Comment (moolen) [inline docs/api/controller-options.md:62]:
```suggestion
| `--zap-time-encoding`                                  | string   | epoch                          | time encoding to use, one of: epoch, millis, nano, iso8601, rfc3339, rfc3339nano                                                                                            |
```

**Review by moolen — APPROVED** [review]

**Review by moolen — APPROVED** [review]

**Comment by moolen** [comment]
:tada: LGTM, thank you for this!

**Pull request merged**